cmd/kubelet/app/server.go

```diff
@@ -597,6 +622,94 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 	return nil
 }
 
+func updateTransport(clientConfig *restclient.Config, clientCertificateManager certificate.Manager) error {
+	pool := x509.NewCertPool()
+	if len(clientConfig.CAFile) > 0 {
+		if certs, err := certutil.NewPool(clientConfig.CAFile); err != nil {
+			return fmt.Errorf("unable to load the CA from %q: %v", clientConfig.CAFile, err)
+		} else {
+			pool = certs
+		}
+	}
+	if len(clientConfig.CAData) > 0 {
+		if certs, err := certutil.ParseCertsPEM(clientConfig.CAData); err != nil {
+			return fmt.Errorf("unable to parse the CA from the data in the config file: %v", err)
+		} else {
+			for _, cert := range certs {
+				pool.AddCert(cert)
+			}
+		}
+	}
+	tlsConfig, err := restclient.TLSConfigFor(clientConfig)
+	if err != nil {
+		return fmt.Errorf("unable to configure TLS for the rest client: %v", err)
+	}
+	tlsConfig.RootCAs = pool
+	tlsConfig.Certificates = nil
+	tlsConfig.GetClientCertificate = func(requestInfo *tls.CertificateRequestInfo) (*tls.Certificate, error) {
+		cert := clientCertificateManager.Current()
+		if cert == nil {
+			return nil, fmt.Errorf("no certificate available")
+		}
+		return cert, nil
+	}
+	clientConfig.Transport = utilnet.SetTransportDefaults(&http.Transport{
```

Comment:
these are different than what you would have gotten otherwise: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/transport/cache.go#L67

it's hard to tell how important the differences are